src/uu/link/src/link.rs

```diff
@@ -8,14 +8,42 @@
 #[macro_use]
 extern crate uucore;
 
+use clap::{App, Arg, ArgMatches};
 use std::fs::hard_link;
 use std::io::Error;
-use std::path::Path;
-use uucore::InvalidEncodingHandling;
+use std::path::{Path, PathBuf};
 
-static SYNTAX: &str = "[OPTIONS] FILE1 FILE2";
-static SUMMARY: &str = "Create a link named FILE2 to FILE1";
-static LONG_HELP: &str = "";
+static VERSION: &str = env!("CARGO_PKG_VERSION");
+static ABOUT: &str = "Call the link function to create a link named FILE2 to an existing FILE1.";
+
+pub mod options {
+    pub static FILE1: &str = "FILE1";
+    pub static FILE2: &str = "FILE2";
+}
+
+#[derive(Debug)]
+struct Options {
+    old: PathBuf,
+    new: PathBuf,
+}
+
+impl Options {
+    fn new(matches: &ArgMatches) -> Options {
+        let old = match matches.value_of_os(options::FILE1) {
+            Some(s) => Path::new(s).to_owned(),
+            None => crash!(1, "{}", msg_wrong_number_of_arguments!(2)),
+        };
+        let new = match matches.value_of_os(options::FILE2) {
+            Some(s) => Path::new(s).to_owned(),
+            None => crash!(1, "{}", msg_wrong_number_of_arguments!(2)),
+        };
+        Options { old, new }
+    }
+}
+
+fn get_usage() -> String {
+    format!("{0} FILE1 FILE2\tor: {0} OPTION", executable!())
```

Comment:
GNU `link` uses `\n` instead of `\t`

Reply:
on a closer look, it looks like `uu_link` doesn't support `link OPTION`

Reply:
`link OPTION` is now removed.